openforcefield/typing/engines/smirnoff/parameters.py

```diff
@@ -2356,6 +2448,21 @@ def create_force(self, system, topology, **kwargs):
         # Add all bonds to the system.
         bond_matches = self.find_matches(topology)
 
+        # Determine if any parameters rely on bond order interpolation, but
+        # only if they were found as matches
+        uses_interpolation = False
+        for match in bond_matches.values():
+            if match.parameter_type.k_bondorder:
+                uses_interpolation = True
+                break
+
+        # Re-computing fractional bond orders is expensive and should be skipped if not necessary
+        if uses_interpolation:
+            for top_mol in topology.topology_molecules:
+                top_mol.reference_molecule.assign_fractional_bond_orders(
+                    bond_order_model=self.fractional_bondorder_method.lower()
```

Comment:
Might be better to do the `.lower()` downstream, i.e. to avoid 

```
In [4]: mol.assign_fractional_bond_orders('AM1-Wiberg')
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-4-8eb006fac6a1> in <module>
----> 1 mol.assign_fractional_bond_orders('AM1-Wiberg')

~/software/openforcefield/openforcefield/topology/molecule.py in assign_fractional_bond_orders(self, bond_order_model, toolkit_registry, use_conformers)
   2644                 self,
   2645                 bond_order_model=bond_order_model,
-> 2646                 use_conformers=use_conformers,
   2647             )
   2648         elif isinstance(toolkit_registry, ToolkitWrapper):

~/software/openforcefield/openforcefield/utils/toolkits.py in call(self, method_name, raise_exception_types, *args, **kwargs)
   4965                     for exception_type in raise_exception_types:
   4966                         if isinstance(e, exception_type):
-> 4967                             raise e
   4968                     errors.append((toolkit, e))
   4969

~/software/openforcefield/openforcefield/utils/toolkits.py in call(self, method_name, raise_exception_types, *args, **kwargs)
   4961                 method = getattr(toolkit, method_name)
   4962                 try:
-> 4963                     return method(*args, **kwargs)
   4964                 except Exception as e:
   4965                     for exception_type in raise_exception_types:

~/software/openforcefield/openforcefield/utils/toolkits.py in assign_fractional_bond_orders(self, molecule, bond_order_model, use_conformers)
   2278         else:
   2279             raise ValueError(
-> 2280                 f"Bond order model '{bond_order_model}' is not supported by OpenEyeToolkitWrapper. "
   2281                 f"Supported models are ['am1-wiberg', 'pm3-wiberg']"
   2282             )

ValueError: Bond order model 'AM1-Wiberg' is not supported by OpenEyeToolkitWrapper. Supported models are ['am1-wiberg', 'pm3-wiberg']
```